packages/flutter/lib/src/painting/edge_insets.dart

```diff
@@ -859,7 +859,7 @@ class _MixedEdgeInsets extends EdgeInsetsGeometry {
       case TextDirection.ltr:
         return new EdgeInsets.fromLTRB(_start + _left, _top, _end + _right, _bottom);
       case TextDirection.rtl:
-        return new EdgeInsets.fromLTRB(_end + _left, _top, _start + _left, _bottom);
+        return new EdgeInsets.fromLTRB(_end + _left, _top, _start + _right, _bottom);
```

Comment:
feel free to flip the order of switch statements like this when they are ltr first. I think we should do as you suggest and have them all rtl first.